### ISSUE TITLE
Mission provider disappear before deadline

### ISSUE THREAD

**Issue by DynV** [pioneerspacesim/pioneer#5442]
### Observed behaviour
If one agree to the out-of-fuel mission from the station of the attached savefile (so before undocking), then go toward the ship in need of help, the one that provided the mission, as fast as one can, it will disappear before the deadline.

### Expected behaviour
The ship in need of help should keep existing until the deadline.

### Steps to reproduce

1. agree to the out-of-fuel mission
2. undock
3. go toward the ship related to that mission as fast as you can
4. it disappears before the mission deadline

__My pioneer version (and OS):__
20220203-win

The attached file was simply renamed so it would go through the upload extension filter, there was no modification to the file content of my save-file.
[Pioneer-Yoan_Faure-230609-Rosu.zip](https://github.com/pioneerspacesim/pioneer/files/10235136/Pioneer-Yoan_Faure-230609-Rosu.zip)

**Comment by Web-eWorks** [comment]
This has most likely been resolved by #5446. Closing; please re-open if you can reproduce the issue on current master.

**Comment by DynV** [comment]
I'm not sure what happened but I loaded the gamesave and was able to reach the target with time to spare. Before I would spawn on the other side of the planet (that the station was orbiting), and it would disappear--much--sooner than the mission deadline. I'm not sure if it was before or after I created that gamesave, but I did switch to the current master then back; I think it was after, so perhaps the reinstall fixed the issue.
